src/x509/clu_x509_sign.c

```diff
@@ -599,25 +599,22 @@ int wolfCLU_CertSign(WOLFCLU_CERT_SIGN* csign, WOLFSSL_X509* x509)
     }
 
     /* set extensions */
-    if (csign->ext != NULL) {
+    if (ret == WOLFCLU_SUCCESS && csign->ext != NULL) {
         wolfCLU_setExtensions(x509, csign->config, csign->ext);
     }
 
     /* sign the certificate */
-    if (csign->keyType == RSAk || csign->keyType == ECDSAk) {
-        if (ret == WOLFCLU_SUCCESS) {
-            if (wolfSSL_X509_check_private_key(csign->ca, csign->caKey.pkey) !=
-                    WOLFSSL_SUCCESS) {
-                wolfCLU_LogError("Private key does not match with CA");
-                ret = WOLFCLU_FATAL_ERROR;
-            }
+    if (ret == WOLFCLU_SUCCESS &&
+            (csign->keyType == RSAk || csign->keyType == ECDSAk)) {
+        if (wolfSSL_X509_check_private_key(csign->ca, csign->caKey.pkey) !=
+                WOLFSSL_SUCCESS) {
+            wolfCLU_LogError("Private key does not match with CA");
+            ret = WOLFCLU_FATAL_ERROR;
         }
 
-        if (ret == WOLFCLU_SUCCESS) {
```

Comment:
Before this change it looks like there was a check on ret value before calling wolfSSL_X509_sign? Could you add that back in-case wolfSSL_X509_check_private_key fails?

Reply:
Great point